libs/velo-external-db-core/src/web/domain-to-spi-error-translator.ts

```diff
@@ -1,48 +1,56 @@
-import { ErrorMessage } from '../spi-model/errors'
+import { ItemAlreadyExistsError, CollectionNotFoundError, ItemNotFoundError, CollectionAlreadyExistsError, CollectionChangeNotSupportedError, UnknownError, InvalidPropertyError, UnauthorizedError, FieldAlreadyExistsError } from '../spi-model/errors'
 
-export const domainToSpiErrorTranslator = (err: any) => {
+export const domainToSpiErrorTranslator = (err: domainErrors.BaseHttpError) => {
     switch(err.constructor) {
       case domainErrors.ItemAlreadyExists: 
-        const itemAlreadyExists: domainErrors.ItemAlreadyExists = err
-        return ErrorMessage.itemAlreadyExists(itemAlreadyExists.itemId, itemAlreadyExists.collectionName, itemAlreadyExists.message)
+        const itemAlreadyExists = err as domainErrors.ItemAlreadyExists
+        return new ItemAlreadyExistsError(itemAlreadyExists.collectionName, itemAlreadyExists.itemId, itemAlreadyExists.message)
   
       case domainErrors.CollectionDoesNotExists:
-        const collectionDoesNotExists: domainErrors.CollectionDoesNotExists = err
-        return ErrorMessage.collectionNotFound(collectionDoesNotExists.collectionName, collectionDoesNotExists.message)
+        const collectionDoesNotExists = err as domainErrors.CollectionDoesNotExists
+        return new CollectionNotFoundError(collectionDoesNotExists.collectionName, collectionDoesNotExists.message)
       
       case domainErrors.FieldAlreadyExists:
-        const fieldAlreadyExists: domainErrors.FieldAlreadyExists = err
-        return ErrorMessage.itemAlreadyExists(fieldAlreadyExists.fieldName, fieldAlreadyExists.collectionName, fieldAlreadyExists.message)
+        const fieldAlreadyExists = err as domainErrors.FieldAlreadyExists
+        return new FieldAlreadyExistsError(fieldAlreadyExists.collectionName, fieldAlreadyExists.fieldName, fieldAlreadyExists.message)
       
       case domainErrors.FieldDoesNotExist:
-        const fieldDoesNotExist: domainErrors.FieldDoesNotExist = err
-        return ErrorMessage.invalidProperty(fieldDoesNotExist.collectionName, fieldDoesNotExist.propertyName, fieldDoesNotExist.message)
+        const fieldDoesNotExist = err as domainErrors.FieldDoesNotExist
+        return new InvalidPropertyError(fieldDoesNotExist.collectionName, fieldDoesNotExist.propertyName, fieldDoesNotExist.message)
   
-      case domainErrors.UnsupportedSchemaOperation:
-        const unsupportedSchemaOperation: domainErrors.UnsupportedSchemaOperation = err
-        return ErrorMessage.operationIsNotSupportedByCollection(unsupportedSchemaOperation.collectionName, unsupportedSchemaOperation.operation, unsupportedSchemaOperation.message)
+      // case domainErrors.UnsupportedSchemaOperation:
+      //   const unsupportedSchemaOperation = err as domainErrors.UnsupportedSchemaOperation
+      //   return ErrorMessage.operationIsNotSupportedByCollection(unsupportedSchemaOperation.collectionName, unsupportedSchemaOperation.operation, unsupportedSchemaOperation.message)
```

Comment:
There is no Wix error for such an error, and I'm not sure it's necessary.
What do you think?